### PR TITLE
ICP-4838 Send default unlock event for Samsung locks

### PR DESCRIPTION
**Pull request by dkirker**

### CODE REVIEW

**Review by ALamchaS — DISMISSED** [review]

File: devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy [hunk 1]
```diff
@@ -1704,6 +1708,21 @@ def isYaleLock() {
 	return false
 }
 
+/**
+ * Utility function to check if the lock manufacturer is Samsung
+ *
+ * @return true if the lock manufacturer is Samsung, else false
+ */
+private isSamsungLock() {
+	if ("022E" == zwaveInfo.mfr) {
```
Comment (greens) [inline devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy:1717]: Do you think we should add a null check for `zwaveInfo`?

Comment (mckeed) [reply devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy:1717]: getZwaveInfo() returns `[:]` for the null case

File: devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy [hunk 2]
```diff
@@ -103,6 +103,10 @@ def installed() {
 	// Device-Watch pings if no device events received for 1 hour (checkInterval)
 	sendEvent(name: "checkInterval", value: 1 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
 
+	if (isSamsungLock()) { // Samsung locks won't pair when locked, so it must be unlocked
```
Comment (kianooshkarami) [inline devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy]: Won't pair, what does that mean in respect to Z-Wave? Secure Inclusion?

Comment (dkirker) [reply devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy]: @kianooshST Sorry, the comment was a little vague. I updated it in the code. Basically, you can't enter the pairing menu if the lock is locked. The idea being how can you be inside and outside at the same time. :P

**Comment by dkirker** [comment]
@ALamchaS Heh, so I updated a comment and it dismissed your review... :/

**Comment by ALamchaS** [comment]
No problem. I don't have write permissions anyway so dismissing my review just changes one icon.

**Review by ALamchaS — APPROVED** [review]

**Review by mckeed — APPROVED** [review]

**Review by greens — APPROVED** [review]

**Pull request merged**